File: liqp-core/src/main/java/liqp/filters/Default.java
package liqp.filters;

import liqp.TruthKt;
import liqp.nodes.RenderContext;

public class Default extends Filter {

  @Override
  public Object apply(RenderContext context, Object value, Object... params) {

    if (params == null || params.length == 0) {
      return value;
    }

    if (TruthKt.isFalsy(value)) {
      return params[0];
    }

    return value;
  }
}

File: liqp-core/src/main/kotlin/liqp/filter/Default.kt
package liqp.filter

import liqp.context.LContext
import liqp.isFalsy
import liqp.params.FilterParams

class Default : LFilter() {

  override fun onFilterAction(context: LContext, value: Any?, params: FilterParams): Any? {
    return when {
      value.isFalsy()-> params[0]
      else-> value
    }
  }
}
